### PR TITLE
onEmptyFilter selection for automations

### PR DESCRIPTION
**Pull request by shogunpurple**
## Description
This PR contains code that allows you to specify the behaviour of the query rows automation step whenever a `null` filter is encountered. The reasoning for this is that couch will return all results of the table if you filter on a value of `null`, rather than nothing. This is particularly dangerous if you plan to delete rows or perform actions on a loop. 

**Empty Filter Setting**
I've opted for a "When Filter Empty" option in the query rows automation. I felt this was more descriptive and clear than just a checkbox.

Screenshot 2022-08-10 at 15 52 40

The 2 options work as follows:

**Return all table rows**
When a user passes no filters, or a value is null, just retain the old behaviour. This maintains backwards compatibility and still lets you account for use cases such as `not equals null`. 

**Return no rows**
When no filters are passed, or any filter is passed that has a `null` value - simply return an empty array, and a success value of `false`. This is a safeguard for users who want to ensure that all their filters are populated before performing a query to the database. This is a good option to use if your next step is `Delete Row` or any other destructive operation. This should probably be the default - but I didn't want to break any backwards compat for existing automation users.

Addresses: 
- https://github.com/Budibase/budibase/issues/6013

## Screenshots
_If a UI facing feature, a short video of the happy path, and some screenshots of the new functionality._

### CODE REVIEW

File: packages/server/src/automations/steps/queryRows.js [hunk 1]
```diff
@@ -103,6 +119,9 @@ function typeCoercion(filters, table) {
   return filters
 }
 
+const hasNullFilters = filters =>
+  filters.length === 0 || filters.some(filter => filter.value === null)
```
Comment (aptkingston) [inline packages/server/src/automations/steps/queryRows.js:123]: The main reason for the current behaviour is that empty strings are the biggest problem - so those are important ones to filter out here. We should update this to `filter.value == null || filter.value === ""` to ensure we account for those.

**Review by aptkingston — COMMENTED** [review]
Just one change needed to account for empty strings but otherwise looks good :+1:

File: packages/server/src/automations/tests/queryRows.spec.js [hunk 2]
```diff
@@ -48,4 +48,71 @@ describe("Test a query step automation", () => {
     expect(res.rows.length).toBe(2)
     expect(res.rows[0].name).toBe(NAME)
   })
+
+  it("Returns all rows when onEmptyFilter has no value and no filters are passed", async () => {
+    console.log(rows)
```
Comment (melohagan) [inline packages/server/src/automations/tests/queryRows.spec.js:53]: console log

File: packages/server/src/automations/steps/queryRows.js [hunk 3]
```diff
@@ -127,9 +146,21 @@ exports.run = async function ({ inputs, appId }) {
...
+
+    if (
+      inputs.onEmptyFilter === EmptyFilterOptions.RETURN_NONE &&
+      inputs["filters-def"] &&
+      hasNullFilters(inputs["filters-def"])
+    ) {
+      rows = []
+    } else {
+      await rowController.search(ctx)
+      rows = ctx.body ? ctx.body.rows : []
```
Comment (melohagan) [inline packages/server/src/automations/steps/queryRows.js:159]: could this be `ctx.body?.rows || []`

**Review by melohagan — APPROVED** [review]

File: packages/server/src/automations/steps/queryRows.js (hunk 3 above)
Comment (shogunpurple) [reply packages/server/src/automations/steps/queryRows.js:160]: Unfortunately we can't use `body?` on the backend due to our node version

**Pull request merged**